### PR TITLE
✨(bbb-stress-test) initial working version of the stress testing tool

### PR DESCRIPTION
**Pull request by madmatah**
Initial version of the stress testing tool based on Chromedriver and selenium. The stress test can be launched via docker-compose. See README.md for more details.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+WORKDIR /builder
+
+# Copy required python dependencies
+COPY setup.py setup.cfg MANIFEST.in /builder/
+COPY ./src/bbb_stress_test /builder/src/bbb_stress_test/
+
+RUN mkdir /install && \
+    pip install --prefix=/install .
+
+RUN ls -lR /install
```
Comment (lunika) [inline Dockerfile:20]: This debug step can maybe removed ?

File: README.md [hunk 2]
```diff
@@ -4,6 +4,36 @@
 
 This is a stress testing tool for [BigBlueButton](https://bigbluebutton.org/).
 
+It simulates client activity (listening only, with microphone and/or webcam) thanks to [Chromedriver](https://chromedriver.chromium.org/) and [Selenium Python bindings](https://selenium-python.readthedocs.io/).
+Each client is a browser running in a virtual framebuffer (with `xvfb`), launched inside a [Docker](http://docker.io/) container.
```
Comment (lunika) [inline README.md:8]:
```suggestion
Each client is a browser running in a virtual framebuffer (with `xvfb`), launched inside a [Docker](https://www.docker.com/) container.
```

File: README.md [hunk 3]
```diff
@@ -4,6 +4,36 @@
...
+
+
+## Getting Started
+
+### Preparation
+
+1) Clone this repository
+
+2) Run `make .env` and update the `.env` file to specify `BBB_URL` and `BBB_SECRET`.\
+You can get these values by running `bbb-conf --secret` on your BBB server.
```
Comment (lunika) [inline README.md:20]: `make build` or `make bootstrap` is missing here no ?

File: setup.cfg [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    Intended Audience :: Developers
+    License :: OSI Approved :: MIT License
+    Natural Language :: English
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3.8
+
+[options]
+include_package_data = True
+install_requires =
+		selenium==3.141.0
```
Comment (lunika) [inline setup.cfg:23]: There is an indentation issue here I think

File: src/bbb_stress_test/client.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        options.add_argument(f"--use-file-for-fake-audio-capture={args.audioInput}")
+    options.add_argument("--mute-audio")
+
+    logging.info("Starting browser")
+
+    return webdriver.Chrome(executable_path="chromedriver", options=options)
+
+
+def join_bbb_meeting(args, browser, join_url):
+    """Control the browser to joint the meeting."""
```
Comment (lunika) [inline src/bbb_stress_test/client.py:39]:
```suggestion
    """Control the browser to join the meeting."""
```

File: Dockerfile (hunk 1 above)
Comment (madmatah) [reply Dockerfile]: Oops :man_facepalming:

File: README.md (hunk 3 above)
Comment (madmatah) [reply README.md:20]: The `bootstrap` and `build` target are for development purpose only, so I did'n mentionned them yet in the README.md.

The Getting started section describe how to start a test with the docker image published on DockerHub.

**Comment by madmatah** [comment]
Thanks @lunika for your feedback :pray:  I made a fixup commit.

**Comment by madmatah** [comment]
I give up this POC with python / selenium /docker because it has very poor performance :cry: 

I'm working on a better version with puppeteer, I'll submit a PR soon.

**Comment by madmatah** [comment]
Replaced by #2

**Pull request closed without merging**